### PR TITLE
fix(auth): allow long BCP-47 language codes in RecaptchaVerifier

### LINKED ISSUE #8357: ` isHostLanguageValid()` in auth's recaptcha loader incorrectly determines long language codes to be invalid
### Operating System

Windows 11

### Browser Version

Chrome/126.0.6478.127

### Firebase SDK Version

10.12.3

### Firebase SDK Product:

Auth

### Describe your project's tooling

NextJS 15 RC
React Compiler

### Describe the problem

My browser's language is `en-GB-oxendict` (got by doing `navigator.languages[0]`).

When trying to use `auth.useDeviceLanguage()`, which sets the above value to `auth.languageCode`, the recaptcha loader will fail an assertion as described [here](https://github.com/firebase/firebase-js-sdk/blob/52f8deb1dc07cd6acc323f36d8be158e927d17c6/packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts#L59), & as implemented [here](https://github.com/firebase/firebase-js-sdk/blob/52f8deb1dc07cd6acc323f36d8be158e927d17c6/packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts#L127), throwing a `FirebaseError: Firebase: Error (auth/argument-error)`.

A bit perculiar that language codes are expected to be below 6 characters. Either that, or `auth.useDeviceLanguage()` should handle it.

### Steps and code to reproduce issue

```ts
import { getAuth } from 'firebase/auth';
import { app } from './app';

export const auth = getAuth(app);
// either, on a environment with a long language string:
// auth.useDeviceLanguage();
// or, set it directly here:
auth.languageCode = 'en-GB-oxendict';

const el = document.createElement('div');
document.body.appendChild(el);

const verifier = new RecaptchaVerifier(auth, el, {
	size: 'invisible',
});
void verifier.render().then(console.log);
// uncaught promise rejection:
// FirebaseError: Firebase: Error (auth/argument-error).
//    at createErrorInternal (index-54738136.js:455:37)
//    at _assert (index-54738136.js:461:11)
//    at ReCaptchaLoaderImpl.load (index-54738136.js:8134:5)
//    at RecaptchaVerifier.init (index-54738136.js:8382:50)
//    at async RecaptchaVerifier.makeRenderPromise (index-54738136.js:8367:5)
```

### PR DESCRIPTION
**Pull request by ATKasem**
### Discussion

Fixes #8357.

`isHostLanguageValid` in the reCAPTCHA loader (`packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts`) rejects any host-language code longer than 6 characters:

```ts
function isHostLanguageValid(hl: string): boolean {
  return hl.length <= 6 && /^\s*[a-zA-Z0-9\-]*\s*$/.test(hl);
}
```

So a valid BCP-47 language tag such as `en-GB-oxendict` (what `navigator.language` can return, and what `auth.useDeviceLanguage()` then sets) fails the assertion in `load()`, and `RecaptchaVerifier.render()` rejects with `auth/argument-error`.

The character-class check (`[a-zA-Z0-9-]`) already keeps the value safe to interpolate into the reCAPTCHA script URL — the length limit is only a sanity bound. This raises it to 30 so valid language tags are accepted while keeping the parameter bounded. The change is purely additive: no value that was previously accepted becomes invalid.

### Testing

Added a regression test in `recaptcha_loader.test.ts` that loads reCAPTCHA with `hl = 'en-GB-oxendict'` and asserts it resolves. It fails before the change (`auth/argument-error`) and passes after; the existing invalid-host-language (injection) test still passes. Verified with `karma start --unit` (browser unit tests). ESLint is clean and a changeset (`@firebase/auth` patch) is included.

### API Changes

None.

### CODE REVIEW

**Comment by ATKasem** [comment]
@sam-gc @pashanka — gentle ping, this has been open since July 11.

Small fix (+18/-1): `isHostLanguageValid` caps host-language codes at 6 characters, so a valid BCP-47 tag like `en-GB-oxendict` — which `navigator.language` can return and `useDeviceLanguage()` then sets — makes `RecaptchaVerifier.render()` reject with `auth/argument-error`. That's #8357.

The CLA check passes and CI is green. Happy to adjust if you'd prefer a different bound, or normalization instead of relaxing the length limit.

File: packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts [hunk 1]
```diff
@@ -125,7 +125,11 @@ export class ReCaptchaLoaderImpl implements ReCaptchaLoader {
 }
 
 function isHostLanguageValid(hl: string): boolean {
-  return hl.length <= 6 && /^\s*[a-zA-Z0-9\-]*\s*$/.test(hl);
+  // The character class keeps the value safe to interpolate into the reCAPTCHA
+  // script URL. The length limit is only a sanity bound and must be permissive
+  // enough for valid BCP-47 language tags such as `en-GB-oxendict`, which the
+  // previous limit of 6 characters incorrectly rejected (#8357).
+  return hl.length <= 30 && /^\s*[a-zA-Z0-9\-]*\s*$/.test(hl);
```
[the remark this answers is not in the document]
Comment (macastelaz) [reply packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts]: I'd say since we are already updating this, lets go for 35 here and point to https://datatracker.ietf.org/doc/html/rfc5646#section-4.4.1 as the justification for this value.

File: packages/auth/src/platform_browser/recaptcha/recaptcha_loader.test.ts [hunk 2]
```diff
@@ -149,6 +149,14 @@ describe('platform_browser/recaptcha/recaptcha_loader', () => {
           'auth/argument-error'
         );
       });
+
+      it('accepts a long BCP-47 host language (regression #8357)', async () => {
```
Comment (macastelaz) [inline packages/auth/src/platform_browser/recaptcha/recaptcha_loader.test.ts]: nit: I don't think it's necessary to include the "why" of a test like this and we can remove the "(regression #8357)" from the test case name. 

Given these tests are functional tests it is sufficient to name the test case in a way that simply describes the expected functionality.

Comment (macastelaz) [inline packages/auth/src/platform_browser/recaptcha/recaptcha_loader.test.ts]: Should we also test other boundary conditions (e.g. exactly 35 characters, 36 characters - assuming we update the impl to use the 35 char limit).

**Review by macastelaz — COMMENTED** [review]
Thanks for the contribution here!

File: packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ATKasem) [reply packages/auth/src/platform_browser/recaptcha/recaptcha_loader.ts]: Done — the limit is now 35, with the RFC cited inline as the justification:

```ts
// The character class keeps the value safe to interpolate into the reCAPTCHA
// script URL. The length limit is only a sanity bound: RFC 5646 (BCP 47)
// section 4.4.1 states that protocols specifying a limited buffer size for
// language tags "MUST allow for language tags of at least 35 characters".
// https://datatracker.ietf.org/doc/html/rfc5646#section-4.4.1
return hl.length <= 35 && /^\s*[a-zA-Z0-9\-]*\s*$/.test(hl);
```

File: packages/auth/src/platform_browser/recaptcha/recaptcha_loader.test.ts (hunk 2 above)
Comment (ATKasem) [reply packages/auth/src/platform_browser/recaptcha/recaptcha_loader.test.ts]: Agreed — renamed to `accepts a long BCP-47 host language`. The issue number is still in the git history and the changeset, so nothing is lost.

Comment (ATKasem) [reply packages/auth/src/platform_browser/recaptcha/recaptcha_loader.test.ts]: Added both boundaries:

```ts
it('accepts a host language of exactly 35 characters', async () => {
  const hl = 'en-Latn-GB-oxendict-u-nu-latn-x-prv';
  expect(hl).to.have.lengthOf(35);
  ...
});

it('fails if the host language exceeds 35 characters', async () => {
  const hl = 'en-Latn-GB-oxendict-u-nu-latn-x-priv';
  expect(hl).to.have.lengthOf(36);
  ...
});
```

The two tags are the same string one character apart, so the pair brackets the limit exactly. I asserted the lengths inline so the literals can't silently drift away from the boundary they are meant to pin.

Verified in headless Chrome — all 12 tests in the file pass. As a sanity check I also temporarily reverted the limit to 30 and re-ran: the "exactly 35 characters" case fails, confirming the new test actually pins the behavior rather than passing vacuously.
